File: cCode/Sorting/quicksort.c
#include<stdio.h>
#include<conio.h>
void quicksort(int [],int,int);
int partition(int [],int,int);
void swap(int *,int *);
void main(){
	int n,i;
	printf("How many numbers do you want to sort: ");
	scanf("%d",&n);
	int rawdata[n];
	printf("Start entering the elements:\n");
	for(i=0;i<n;i++)
		scanf("%d",&rawdata[i]);
	quicksort(rawdata,0,n-1);
	printf("Sorted elements are:\n");
	for(i=0;i<n;i++)
		printf("%d\t",rawdata[i]);
	getch();
}
void quicksort(int raw[],int beg,int end){
	int loc;
	if(beg<end){
		loc=partition(raw,beg,end);
		quicksort(raw,beg,loc-1);
		quicksort(raw,loc+1,end);
	}
}
int partition(int arr[],int big,int end){
	int i=big-1,j,x=arr[end];
	for(j=big;j<=end-1;j++){
		if(arr[j]<=x){
			i++;
			swap(&arr[i],&arr[j]);
		}
	}
	swap(&arr[i+1],&arr[end]);
	return (i+1);
}
void swap(int *a, int *b){
	int temp;
	temp = *a;
  *a   = *b;
  *b   = temp;
}

File: javaCode/DataStructure/sorting/quickSort.java
package sorting;
public class quickSort
{
	public int partition(int A[],int beg,int end)
	{
		int loc=beg,i=beg,j=end,temp;
		boolean flag=false;
		while(flag!=true)
		{
			while(A[loc]<=A[j] && loc!=j)	j--;
			if(loc==j)	flag=true;
			else
			{
				temp=A[loc];
				A[loc]=A[j];
				A[j]=temp;
				loc=j;
			}
			if(flag!=true)
			{
				while(A[loc]>=A[i] && loc!=i)	i++;
				if(loc==i)	flag=true;
				else
				{
					temp=A[loc];
					A[loc]=A[i];
					A[i]=temp;
					loc=i;
				}
			}
		}
		return loc;
	}
	public void qckSort(int A[],int p,int r)
	{
		int pivot;
		if(p<r)
		{
			pivot=partition(A,p,r);
			qckSort(A,p,pivot-1);
			qckSort(A,pivot+1,r);
		}
	}
}
